lib/ssl/ssl3con.c

```diff
@@ -10371,15 +9664,14 @@ ssl3_HandleCertificateVerify(sslSocket *ss, SSL3Opaque *b, PRUint32 length,
         goto alert_loser;
     }
 
-    if (ss->ssl3.hs.hashType != handshake_hash_record) {
-        if (!hashes) {
-            PORT_Assert(0);
-            desc = internal_error;
-            errCode = SEC_ERROR_LIBRARY_FAILURE;
-            goto alert_loser;
-        }
-        hashesForVerify = hashes;
-    } else {
+    if (!hashes) {
```

Comment:
I was trying to say that ssl3_HandleCertificateVerify() will be called when we receive a certificate_verify message but don't expect any. In that case `hashes == NULL`. So we'd better assert after we checked that we actually expect a certificate_verify message, otherwise we'll throw a library error when we shouldn't.